### PR TITLE
feat: add find_runtime_files

### PR DESCRIPTION
**Pull request by kylo252**
Search for file, grep for text or jump to any directory from `vim.opt.runtimepath`.

Inspired by https://github.com/tpope/vim-scriptease

- Default keymaps:
 - `<cr>` or `<C-f>`: search for files in selectied directory
 - `<C-g>`: grep for text in selectied directory
 - `<C-y>`: change CWD to selectied directory

### CODE REVIEW

File: lua/telescope/builtin/internal.lua [hunk 1]
```diff
@@ -1237,4 +1237,57 @@ local function apply_checks(mod)
...
+
+  for _, entry in ipairs(runtimepath) do
+    vim.list_extend(runtimedirs, vim.fn.globpath(entry, "", 1, 1))
+  end
+
+  local find_in_dir = function()
+    local entry = action_state.get_selected_entry()
+    opts.cwd = entry.value
+    files.find_files(opts)
+    vim.cmd [[normal! A]]
```
[the remark this answers is not in the document]
Comment (fdschmidt93) [reply lua/telescope/builtin/internal.lua:1251]: Apologies for the late feedback.

Sorry, the video and screenshots are not straightforward to follow. Easier to push the current state of things and then we can talk about it a bit more transparently :)

```lua
  local grep_in_dir = function()
    local entry = action_state.get_selected_entry()
    opts.cwd = entry.value
    local current_picker = action_state.get_current_picker(prompt_bufnr) 
    actions._close(prompt_bufnr, current_picker.inital_mode == "insert")
    files.live_grep(opts)
```
This does not work? It very closely mimics

https://github.com/nvim-telescope/telescope.nvim/blob/0caec3d6e4d3c3c71339eb18a9aae7ed0f24badc/lua/telescope/builtin/internal.lua#L149-L155

**Comment by fdschmidt93** [comment]
Agreed, let's not interact with globals here.

Attempt 1 is `actions.run_builtin` adapted which I quite like.

> Create a generic action, but we have to modify the opts which I'm not sure are even preserved..

What about something like this?

```lua
actions.builtin_cb = function(prompt_bufnr, opts)
  local picker_opts = vim.tbl_deep_extend("keep", { picker_cb = nil, entry_cb = nil }, opts)
  picker_opts = opts.entry_cb(picker_opts, prompt_bufnr)
  actions._close(prompt_bufnr, true)
  if string.match(opts.picker_cb, " : ") then
    -- Call appropriate function from extensions
    local split_string = vim.split(name, " : ")
    local ext = split_string[1]
    local func = split_string[2]
    require("telescope").extensions[ext][func](picker_opts)
  else
    -- Call appropriate telescope builtin
    require("telescope.builtin")[opts.cb_picker](picker_opts)
  end
end
```

where `picker_cb` would be `{live_grep, find_files}` and `entry_cb` would be something akin to

```lua
function(picker_opts, prompt_bufnr)
  picker_opts.cwd = action_state.get_selected_entry().value
  return opts
end
```

**Comment by kylo252** [comment]
@fdschmidt93, I ended up merging all the changes into `run_builtin` because I didn't like having two nearly identical functions. I'm pretty happy with the current implementation, and it's easy for others to replicate. 

I'm not too good with closures so I couldn't figure out how to move the specific `find_in_dir` action, so let me know if it can be improved. However, as I said, it's now easy to create a new one and this should open the door for quite a few interesting implementations!

**Comment by Conni2461** [comment]
I dont think this is a good fit for telescope core. Its too gimmicky.

**Pull request closed without merging**